Core/Net/NetFileCache.cs

```diff
@@ -96,6 +111,54 @@ public string GetCachePath()
             return cachePath;
         }
 
+        /// <summary>
+        /// Gets the cache path from the system registry.
+        /// </summary>
+        /// <returns></returns>
+        public string GetCachePathFromRegistry()
+        {
+            // Create a new instance of the registry and request the cache path
+            Win32Registry registry = new Win32Registry();
+            return registry.GetCachePath();
+        }
+
+        /// <summary>
+        /// Gets the system's default cache path.
+        /// </summary>
+        /// <returns></returns>
+        public string GetSystemDefaultCachePath()
+        {
+            string baseDirectory;
+
+            // Check which platform we're on, as the path depends on it
+            if (!Platform.IsWindows)
+            {
+                // We are on Linux or OSX, attempt to use the XDG standard for getting the path
+                baseDirectory = Environment.GetEnvironmentVariable("XDG_CACHE_HOME");
+
+                // If the above didn't work, fall back to "~/.local/"
+                if (string.IsNullOrWhiteSpace(baseDirectory))
+                {
+                    baseDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".local", "share");
```

Comment:
This is the same as `SpecialFolder.LocalApplicationData` on Linux as far as I can tell. Opportunity to consolidate with below code here.